akka-actor/src/main/scala/akka/actor/dungeon/TimerSchedulerImpl.scala

```diff
@@ -46,7 +48,12 @@ import akka.util.OptionVal
     }
     val nextGen = nextTimerGen()
 
-    val timerMsg = TimerMsg(key, nextGen, this)
+    val timerMsg =
+      if (msg.isInstanceOf[NotInfluenceReceiveTimeout])
+        NotInfluenceReceiveTimeoutTimerMsg(key, nextGen, this)
+      else
+        TimerMsg(key, nextGen, this)
+
```

Comment:
there is also another timer implementation in akka-actor-typed, which probably suffers from the same problem

Reply:
Fixed that as well. 

I wondering where I should added tests for the changes in akka-actor-typed. It seems like ReceiveTimeouts are previously untested there.

Reply:
@patriknw Hey Patrik, let me know if you have any ideas around tests to add into akka-actor-typed for ReceiveTimeouts. Otherwise I think the changes are complete.